ts/Series/Bubble/BubbleSeries.ts

```diff
@@ -175,6 +175,41 @@ function onAxisFoundExtremes(
 
 }
 
+/**
+ * If a user has defined categories, it is necessary to retroactively hide any
+ * ticks added by the 'onAxisFoundExtremes' function above (#21672).
+ *
+ * Otherwise they can show up on the axis, alongside user-defined categories.
+ */
+function onAxisAfterRender(this: Axis): void {
+    const {
+            ticks,
+            tickPositions,
+            dataMin = 0,
+            dataMax = 0,
+            categories,
+            options,
+            series
+        } = this,
+        type = options.type;
+
+    if (series.find((s): boolean => s.bubblePadding === true)) {
+        if ((categories && categories.length) || type === 'category') {
```

Comment:
These two if's can be combined into one. Run the `categories?.length` check first because it is cheaper than `find`.